source/main/physics/RigSpawner.cpp

```diff
@@ -1063,112 +1062,84 @@ void RigSpawner::ProcessAirbrake(RigDef::Airbrake & def)
 
 void RigSpawner::ProcessWing(RigDef::Wing & def)
 {
-	SPAWNER_PROFILE_SCOPED();
+    SPAWNER_PROFILE_SCOPED();
 
-    if (! CheckWingLimit(1))
-	{
-		return;
-	}
+    // Perform checks
+    if (! this->CheckWingLimit(1)) { return; }
 
-	std::stringstream wing_name;
-	wing_name << "wing-" << m_rig->truckname << "-" << m_rig->free_wing;
-	std::stringstream wing_obj_name;
-	wing_obj_name << "wingobj-" << m_rig->truckname << "-" << m_rig->free_wing;
+    if ((m_first_wing_index != -1) && (m_rig->wings[m_rig->free_wing - 1].fa == nullptr))

```

Comment:
Initial state is `m_first_wing_index = -1; free_wing = 0;`
If `m_first_wing_index != -1`, there must be at least 1 wing.